N/neper/build_tarballs.jl

```diff
@@ -0,0 +1,55 @@
+using BinaryBuilder, Pkg
+
+name = "neper"
+version = v"4.5.0"
+
+sources = [
+    ArchiveSource("https://github.com/neperfepx/neper/archive/refs/tags/v$(version).tar.gz", "db80dd89e02207e9b056b05fb9fbe493199ce7c3736b2039104c595b4dcd02a9")
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir/neper-*
+mkdir build && cd build
+if [[ ${target} == *-apple-* ]]; then
+    HAVE_OPENMP=0
+else
+    HAVE_OPENMP=1
+fi
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DHAVE_OPENMP=${HAVE_OPENMP} ../src
+make -j${nproc}
+make install
+"""
+
+function exclude(p)
+    if get(p.tags, "libc", nothing) == "musl"
+        # src/contrib/ut/src/ut_print/ut_print.c:1685:38: error: parameter 1 (‘beg_time’) has incomplete type
+        #  1685 | ut_print_elapsedtime (struct timeval beg_time, struct timeval end_time)
+        #       |                       ~~~~~~~~~~~~~~~^~~~~~~~
+        return true
+    elseif get(p.tags, "os", nothing) == "windows" && get(p.tags, "arch", nothing) == "i686"
+        # CMake Error at CMakeLists.txt:262 (install):
+        #  install FILES given no DESTINATION!
+        return true
+    elseif get(p.tags, "os", nothing) == "freebsd"
+        # CMake Error at CMakeLists.txt:262 (install):
+        #  install FILES given no DESTINATION!
+        return true
+    end
+    return false
+end
+platforms = supported_platforms(; exclude=exclude)
+platforms = expand_cxxstring_abis(platforms)
+
+products = [
+    ExecutableProduct("neper", :neper)
+]
+
+dependencies = [
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency(PackageSpec(name="GSL_jll", uuid="1b77fbbe-d8ee-58f0-85f9-836ddc23a7a4")),
+    Dependency(PackageSpec(name="NLopt_jll", uuid="079eb43e-fd8e-5478-9966-2cf3e3edb778")),
+    # /opt/aarch64-linux-gnu/bin/../lib/gcc/aarch64-linux-gnu/12.1.0/../../../../aarch64-linux-gnu/bin/ld: /opt/aarch64-linux-gnu/aarch64-linux-gnu/sys-root/usr/local/lib/libscotch.so: undefined reference to `gzwrite'
+    # Dependency(PackageSpec(name="SCOTCH_jll", uuid="a8d0f55d-b80e-548d-aff6-1a04c175f0f9")),
+]
+
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"12.1.0")
```

Comment:
GCC 12 is not going to work on any released version of Julia :slightly_smiling_face:   For maximum compatibility we want to use the _oldest_ version of GCC, not the latest one.

Reply:
Right, I got some errors with gcc 4 and just switch to the latest to get things to run. O am guessing it needs to be 9 (https://github.com/JuliaPackaging/Yggdrasil/blob/master/S/SCOTCH/build_tarballs.jl#L52)

Reply:
You don't _necessarily_ need the same version as the one used for the dependencies, although that's a good initial guess.  It's really a trial and error process.

Reply:
Okay, but in this case neper have a vendored libscotch. I couldn't get it to work when using the JLL dependency as indicated below. Maybe they have some changes to it, or I did something wrong.

Reply:
The error is likely a missing `-lz` in the compiler invocation (I don't see it in the error you shared).  In https://github.com/JuliaPackaging/Yggdrasil/blob/69653bccf74a289e8fa97a548e4dbc294e7b3e30/S/SCOTCH/build_tarballs.jl#L15-L31 it looks like we're using a custom makefile which likely doesn't produce a pkgconfig file, so this build system can't know that libscotch needs linking to zlib as well

Reply:
Just to clarify, you mean that that information is missing from the `SCOTCH_jll` but that I can fix it by adding `-lz` to `neper`s compiler invocation in this PR?